addons/mail/static/src/models/activity/activity.js

```diff
@@ -148,8 +148,13 @@ function factory(dependencies) {
                 method: 'activity_format',
                 args: [this.id],
             }));
-            this.update(this.constructor.convertData(data));
-            this.thread.refresh();
+            if (data) {
+                this.update(this.constructor.convertData(data));
+                this.thread.refreshActivities();
+            } else {
+                this.thread.refresh();
+                this.delete();
```

Comment:
Do we actually need to `refresh` the thread in this case, if we add a manual `delete` now?

cc @xdu-odoo @alexkuhn if you have any idea why this was there

Reply:
I think initially it was to refresh message list as when you close an activity a message can be shown (I think about activities that sends mail for example)

Reply:
![image](https://user-images.githubusercontent.com/48204996/95828083-cafd7200-0d34-11eb-9c89-f2d8bd3237ec.png)

For every type of activity in fact, you have a message to say it's done